src/main/java/com/minecolonies/coremod/entity/ai/mobs/AbstractArcherBarbarian.java

```diff
@@ -0,0 +1,302 @@
+    {
+        if (colony != null)
+        {
+            final int raidLevel = (int) (colony.getRaidLevel() * 1.5);
+            return BARBARIAN_BASE_HEALTH + raidLevel;
+        }
+        return BARBARIAN_BASE_HEALTH;
+    }
+
+    protected void entityInit()
+    {
+        super.entityInit();
+        this.dataManager.register(SWINGING_ARMS, Boolean.FALSE);
+    }
+
+    /**
+     * Get this Entity's EnumCreatureAttribute
+     */
+    public EnumCreatureAttribute getCreatureAttribute()
+    {
+        return EnumCreatureAttribute.UNDEAD;
+    }
+
+    /**
+     * Handles updating while being ridden by an entity
+     */
+    public void updateRidden()
+    {
+        super.updateRidden();
+
+        if (this.getRidingEntity() instanceof EntityCreature)
+        {
+            final EntityCreature entitycreature = (EntityCreature) this.getRidingEntity();
+            this.renderYawOffset = entitycreature.renderYawOffset;
+        }
+    }
+
+    /**
+     * Called when the mob's health reaches 0.
+     */
+    public void onDeath(final DamageSource cause)
+    {
+        super.onDeath(cause);
+
+        if (cause.getSourceOfDamage() instanceof EntityArrow && cause.getEntity() instanceof EntityPlayer)
+        {
+            final EntityPlayer entityplayer = (EntityPlayer) cause.getEntity();
+            final double d0 = entityplayer.posX - this.posX;
+            final double d1 = entityplayer.posZ - this.posZ;
+
+            final double distance = 2500.0D;
+
+            if (d0 * d0 + d1 * d1 >= distance)
+            {
+                entityplayer.addStat(AchievementList.SNIPE_SKELETON);
+            }
+        }
+    }
+
+    /**
+     * Gives armor or weapon for entity based on given DifficultyInstance
+     */
+    protected void setEquipmentBasedOnDifficulty(final DifficultyInstance difficulty)
+    {
+        super.setEquipmentBasedOnDifficulty(difficulty);
+        this.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.BOW));
+    }
+
+    /**
+     * Called only once on an entity when first time spawned, via egg, mob spawner, natural spawning etc, but not called
+     * when entity is reloaded from nbt. Mainly used for initializing attributes and inventory
+     */
+    @Nullable
+    public IEntityLivingData onInitialSpawn(final DifficultyInstance difficulty, @Nullable final IEntityLivingData living)
+    {
+        final IEntityLivingData livingdata = super.onInitialSpawn(difficulty, living);
+        this.setEquipmentBasedOnDifficulty(difficulty);
+        this.setEnchantmentBasedOnDifficulty(difficulty);
+        this.setCombatTask();
+        this.setCanPickUpLoot(this.rand.nextFloat() < 0.55F * difficulty.getClampedAdditionalDifficulty());
+
+        if (this.getItemStackFromSlot(EntityEquipmentSlot.HEAD).isEmpty())
+        {
+            final Calendar calendar = this.world.getCurrentDate();
+
+            if (calendar.get(2) + 1 == 10 && calendar.get(5) == 31 && this.rand.nextFloat() < 0.25F)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 10 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 31 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 0.25F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)